[odh] Sync: Add support for additional fields (#72)

VaishnaviHire opened the pull request:
This commit allows users to include additional fields like annotations, sidecars, labels in server deployment

This change is needed to support adding `otel` sidecar and annotations for observability


Approved-by: leseb

<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Merge criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **Refactor**
  - Improved deployment update process to better preserve existing annotations and labels by using server-side apply for updates.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

VaishnaviHire commented:
@rhuss @leseb Can I get some quick reviews on this and merge it?

rhuss commented:
Sorry, i'm already afk for today but will jump in right tomorrow morning if @leseb is not faster :-)

rhuss commented:
thanks! looks good to me.

VaishnaviHire committed:
Add support for additional fields (#72)

This commit allows users to include additional fields like annotations, sidecars, labels in server deployment

This change is needed to support adding `otel` sidecar and annotations for observability


Approved-by: leseb
